core/templates/dev/head/domain/collection/CollectionRightsBackendApiService.js

```diff
@@ -67,6 +150,22 @@ oppia.factory('CollectionRightsBackendApiService', [
             _setCollectionStatus(
               collectionId, collectionVersion, false, resolve, reject);
           });
+        },
+
+        /**
+         * Returns true if the collection is private given the collectionId.
+         * This method will check the collection rights object exists, if so,
+         * return the result from the cache. However, this will not cache the
+         * collection rights object if it does not exist in the cache.
+         */
+        isCollectionPrivate: function(collectionId) {
```

Comment:
I'm quite concerned about the async part of this method. I tried using promises, but angular didn't like it (digest iteration error). I am leaning towards only allowing checks from the cache (null otherwise).